### PR TITLE
增加同步上传与更新git功能

### PR DESCRIPTION
**Pull request by HelloWorld886**
同步上传笔记到git仓库，从git仓库拉取最新笔记

### CODE REVIEW

**Comment by lovelock** [comment]
不懂C++，但我觉得这个功能很有用，现在我用坚果云同步的，尚可用

File: src/vdirectorytree.cpp [hunk 1]
```diff
@@ -35,10 +35,10 @@ VDirectoryTree::VDirectoryTree(QWidget *parent)
 
     initShortcuts();
 
-    connect(this, SIGNAL(itemExpanded(QTreeWidgetItem*)),
-            this, SLOT(handleItemExpanded(QTreeWidgetItem*)));
-    connect(this, SIGNAL(itemCollapsed(QTreeWidgetItem*)),
-            this, SLOT(handleItemCollapsed(QTreeWidgetItem*)));
+    connect(this, SIGNAL(itemExpanded(QTreeWidgetItem *)),
```
Comment (tamlok) [inline src/vdirectorytree.cpp:38]: It looks like that a format tool is applied for this file? Please revert it to reduce irrelevant changes.

Comment (HelloWorld886) [reply src/vdirectorytree.cpp]: Thank you for your advice, I have corrected my mistake.

File: src/vmainwindow.cpp [hunk 2]
```diff
@@ -990,6 +999,73 @@ void VMainWindow::initMarkdownMenu()
...
+}
+
+void VMainWindow::download()
+{
+    QVector<VNotebook*>& noteBooks = vnote->getNotebooks();
+    for (QVector<VNotebook*>::iterator i = noteBooks.begin(); i < noteBooks.end(); i++)
+    {
+        QString notebookDir = (*i)->getPath();
+        QString notebookName = (*i)->getName();
+		if ((*i)->isOpened())
```
[the remark this answers is not in the document]
Comment (tamlok) [reply src/vmainwindow.cpp]: There are still many places using Tab in the PR.

File: src/utils/vSync.h [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	void uploadFinish();
+
+	void start(const QString &cmd);
+	void showMessageBox(const QString &message, bool showButton);
+	void hideMessageBox();
+	void onMessageButtonClick();
+	QString VSync::getSyncHead(const QString &args) const;
+
+private:
+	QString _dir;
```
Comment (tamlok) [inline src/utils/vSync.h:63]: In VNote, we use `m_xxx` as the name of data members. Anyway, we could leave it unchanged here in this PR.

**Comment by lovelock** [comment]
Why not merge it and give a new release?

**Comment by minli04g** [comment]
求发布

**Pull request merged**